twine/cli.py

```diff
@@ -54,18 +63,10 @@ def dispatch(argv: List[str]) -> Any:
         action="store_true",
         help="disable colored output",
     )
-    parser.add_argument(
-        "command",
-        choices=registered_commands.names,
-    )
-    parser.add_argument(
-        "args",
-        help=argparse.SUPPRESS,
-        nargs=argparse.REMAINDER,
-    )
-
-    parser.parse_args(argv, namespace=args)
-
-    main = registered_commands[args.command].load()
-
-    return main(args.args)
+    sub_parser = parser.add_subparsers(title='sub command to run', dest='command')
+    sub_parser.required = True
+    for command in registered_commands:
+        main_func = command.load()
+        generator = main_func(sub_parser)
+        command_map[command.name] = next(generator)
+    return parser
```

Comment:
This feels like a non-trivial refactoring that's less obvious than the existing implementation.